src/D2L.CodeStyle.Analyzers/Immutability/ImmutableTypeInfo.cs

```diff
@@ -62,17 +62,17 @@ INamedTypeSymbol type
 		) {
 			ImmutableArray<bool> immutableTypeParameters = type
 				.TypeParameters
-				.Select( p => Attributes.Objects.Immutable.IsDefined( p ) )
+				.Select( p => Attributes.Objects.OnlyIf.IsDefined( p ) )

```

Comment:
Only `[OnlyIf]` parameters get considered for `IsImmutableDefinition`.

This means that given
```csharp
[Immutable] IFoo<[Immutable] T> {}
[Immutable] sealed class Holder {
  private readonly IFoo<object> m_field;
}
```

`IFoo<object>` won't be called out as _not being immutable_, because it's an `[Immutable]` type without any conditional parameters. However, there will be another compiler error on `object` itself, since it's not allowed as a parameter to `[Immutable] T`